lib/providers/http/services/aens.js

```diff
@@ -144,14 +179,26 @@ class AENS extends HttpService {
    *
    * @param nameHash
    * @param fee
+   * @param options
    * @returns {Promise<*>}
    */
-  async revoke(nameHash, fee = 1) {
-    let {data} = await this.client.post (
-      'name-revoke-tx',
-      {'name_hash': nameHash, fee},
-      {internal: true})
-    return data['name_hash']
+  async revoke(nameHash, fee = 1, options = {}) {
+    let payload = {'name_hash': nameHash, fee}
+    if (options && options.privateKey) {
+      payload = {
+        ...payload,
+        nonce: options && options.nonce,
+        account: options && options.account || await this.client.accounts.getPublicKey ()
+      }
+      let {tx} = await this.client.post('tx/name/revoke', payload)
+      return await this.client.tx.sendSigned(tx, privateKey)
+    } else {
+      let {data} = await this.client.post (
+        'name-revoke-tx',
+        payload,
+        {internal: true})
+      return data['name_hash']
```

Comment:
Why would you still use the internal api in this case? e.g. why use `name-revoke-tx` instead of `/tx/name/revoke`?